### PR TITLE
core: add support to connect to docker-machine

### LINKED ISSUE #82: Add back support for docker-machine
It requires the use of `dc.NewClientFromEnv()` in `NewPool()`

### PR DESCRIPTION
**Pull request by pmenglund**
Fixes #82 
Signed-off-by: Martin Englund <martin@englund.nu>

### CODE REVIEW

File: dockertest.go [hunk 1]
```diff
@@ -40,19 +41,27 @@ func (r *Resource) GetPort(id string) string {
...
+// variable DOCKER_URL, or from docker-machine if the environment variable DOCKER_MACHINE_NAME is set,
+// or if neither is defined a sensible default for the operating system you are on.
 func NewPool(endpoint string) (*Pool, error) {
+	var client *dc.Client
+	var err error
+
 	if endpoint == "" {
 		if os.Getenv("DOCKER_URL") != "" {
 			endpoint = os.Getenv("DOCKER_URL")
+		} else if os.Getenv("DOCKER_MACHINE_NAME") != "" {
```
Comment (aeneasr) [inline dockertest.go]: Is this env var set per spec on the host running docker machine?

File: dockertest.go [hunk 2]
```diff
@@ -40,19 +41,27 @@ func (r *Resource) GetPort(id string) string {
...
+			client, err = dc.NewClientFromEnv()
+			goto error_check
```
Comment (aeneasr) [inline dockertest.go]: Please handle the error here directly

File: dockertest.go (hunk 1 above)
Comment (pmenglund) [reply dockertest.go]: What do you mean by "per spec"?

You set the environment on your computer using `eval $(docker-machine env default)` when sets the following environment:
```
export DOCKER_TLS_VERIFY="1"
export DOCKER_HOST="tcp://192.168.174.132:2376"
export DOCKER_CERT_PATH="/Users/martin/.docker/machine/machines/default"
export DOCKER_MACHINE_NAME="default"
```

File: dockertest.go (hunk 2 above)
Comment (pmenglund) [reply dockertest.go]: Ok, do you want to keep the goto for the return, or do it directly here too?

File: dockertest.go (hunk 1 above)
Comment (aeneasr) [reply dockertest.go]: I see, thanks for clarifying!

File: dockertest.go (hunk 2 above)
Comment (aeneasr) [reply dockertest.go]: Do it directly here :)

**Comment by pmenglund** [comment]
I don't think I can write a test case for the new code, without some major refactoring as it can't call `dc.NewClientFromEnv()` without having docker-machine running (which isn't available in travis), and setting the environment variables by hand in the test won't work either, as it expects the docker connection to be `tcp://...`

**Comment by aeneasr** [comment]
Ok, thanks!

**Pull request merged**